bleak/backends/bluezdbus/utils.py

```diff
@@ -51,3 +53,53 @@ def extract_service_handle_from_path(path):
         return int(path[-4:], 16)
     except Exception as e:
         raise BleakError(f"Could not parse service handle from path: {path}") from e
+
+
+async def get_default_adapter(bus: MessageBus = None) -> str:
+    """Get name of the first powered Bluetooth adapter
+
+     Args:
+         bus (`MessageBus`): Optional active (connected) D-Bus connection to avoid creating new.
+
+    Returns:
+        Name of the first found powered adapter on the system, available on D-Bus path "/org/bluez/<name>".
+
+    Raises:
+        OSError: if there are no Bluetooth adapters (D-Bus objects implementing org.bluez.Adapter1
+            interface) on the system.
+        OSError: if all found adapters are powered off - use `bluetoothctl power on` to power on the adapter.
+    """
+    bus_provided = bus and bus.connected
+    if not bus_provided:
+        bus = await MessageBus(bus_type=BusType.SYSTEM).connect()
+
+    reply = await bus.call(
+        Message(
+            destination=defs.BLUEZ_SERVICE,
+            path="/",
+            member="GetManagedObjects",
```

Comment:
This is a very expensive operation. Instead of opening a new D-Bus connection and calling this method, could we make use of the existing copy of this information in the BleakScanner?

Reply:
I have added 2 commits (those will be squashed later) which solve this. What do you think? In the afternoon I will also modify `client.py`, which has 50 lines of duplicate code (this and adding signal handlers).